### PR TITLE
releases: add tidb 4.0.15 release notes

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)
releases: add tidb 4.0.15 release notes

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v5.2 (TiDB 5.2 versions)
- [x] v5.1 (TiDB 5.1 versions)
- [x] v5.0 (TiDB 5.0 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/6475
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: releases/release-4.0.15.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+# TiDB 4.0.15 Release Notes
+
+## 兼容性更改
+
+## 功能增强
+
+## 提升改进
+
+## Bug 修复
+
```
Comment (solotzg) [inline releases/release-4.0.15.md:14]: + TiFlash

    - 修复多盘部署时数据不一致的潜在问题
    - 修复当查询过滤条件包含诸如 `CONSTANT` `<` | `<=` | `>` | `>=` `COLUMN` 时出现错误结果的问题
    - 修复写压力重时 metrics 中 store size 不准确的问题
    - 修复 TiFlash 多盘部署时无法恢复数据的潜在问题
    - 修复 TiFlash 长时间运行后无法回收 delta 历史数据的潜在问题

File: releases/release-4.0.15.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
+---
+title: TiDB 4.0.15 Release Notes
+---
+
+# TiDB 4.0.15 Release Notes
+
+## 兼容性更改
+
+## 功能增强
+
```
Comment (disksing) [inline releases/release-4.0.15.md:10]:
```suggestion

+ PD
    - 提升 PD 之间同步 Region 信息的性能 [#3932](https://github.com/tikv/pd/pull/3932)
```

File: releases/release-4.0.15.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+## 提升改进
+
+## Bug 修复
+
++工具
```
Comment (disksing) [inline releases/release-4.0.15.md:15]:
```suggestion

+ PD

    - 修复 PD 不能及时修复 down-peer 的问题 [#4081](https://github.com/tikv/pd/pull/4081)
    - 修复扩容 TiKV 时 PD 可能会 panic 的问题 [#3909](https://github.com/tikv/pd/pull/3909)

+工具
```

File: releases/release-4.0.15.md (hunk 2 above)
Comment (ethercflow) [reply releases/release-4.0.15.md:10]:
```suggestion

+
+ + TiKV
+        - 读写分离减少读取延迟 [#10619](https://github.com/tikv/tikv/pull/10619)
+        - TiKV 协处理器慢日志只会考虑处理请求所花费的时间 
+        - 当 slogger 线程过载且队列已满时，删除日志而不是阻塞线程。 [#10863](https://github.com/tikv/tikv/pull/10863)
+        - 支持动态更改 CDC 配置。 [#10684](https://github.com/tikv/tikv/pull/10684)
+        - 减少已解析的 ts 消息大小以节省网络带宽。[#10677](https://github.com/tikv/tikv/pull/10677)


```

File: releases/release-4.0.15.md (hunk 1 above)
Comment (ethercflow) [reply releases/release-4.0.15.md:14]:
``` suggestion

+ TiKV

    - 修复恢复时启用TDE时br报告文件已存在错误的问题。 [#10917](https://github.com/tikv/tikv/pull/10917)
    - RaftStore Snapshot GC 修复：修复 1 个快照文件 GC 失败时，快照 GC 漏掉 GC 快照文件的问题。[#10871](https://github.com/tikv/tikv/pull/10871)
    - 修复删除陈旧区域过于频繁的问题。[#10781](https://github.com/tikv/tikv/pull/10781)
    - 修复频繁重新连接 pd 客户端。 [#9818](https://github.com/tikv/tikv/pull/9818)
    - 从加密文件字典中检查陈旧的文件信息。  [#10598](https://github.com/tikv/tikv/pull/10598)

```

Comment (guo-shaoge) [reply releases/release-4.0.15.md:14]: ### TiDB
- 修复 binary literal 构造 range 时的 bug 。[26455](https://github.com/pingcap/tidb/pull/26455)
- 修复当查询包含 group by 和 union 时报错 "index out of range" 的问题。[26553](https://github.com/pingcap/tidb/pull/26553)
- 修复有 tombstone 存储时请求发送失败的问题.[25849](https://github.com/pingcap/tidb/pull/25849)
- 修复 case when 表达式字符集相关问题。[26671](https://github.com/pingcap/tidb/pull/26671)
- 修复将非法字符串转为 date 类型时的非预期行为。[27935](https://github.com/pingcap/tidb/pull/27935)
- 修复将 Apply 算子转为 Join 时漏掉 column 信息的问题。[27282](https://github.com/pingcap/tidb/pull/27282)
- 修复 new collation 打开时 count distinct 结果错误问题。[27830](https://github.com/pingcap/tidb/pull/27830)
- 修复 extract 表达式参数为负的 Duration 时结果错误问题。[27369](https://github.com/pingcap/tidb/pull/27369)
- group_concat 聚合函数增加对 collation 信息的处理。[27835](https://github.com/pingcap/tidb/pull/27835)
- 回滚 [#21045](https://github.com/pingcap/tidb/pull/21045) 以避免第一次跑 show variables 很慢的问题。[#24326](https://github.com/pingcap/tidb/issues/24326).[26240](https://github.com/pingcap/tidb/pull/26240)
- 修复 between 表达式类型推导时 collation 错误问题。[27851](https://github.com/pingcap/tidb/pull/27851)
- 修复 greatest(datetime) 表达式 union null 结果错误问题。[26564](https://github.com/pingcap/tidb/pull/26564)
- 修复聚合函数 having 条件被错误地下推的问题。[27741](https://github.com/pingcap/tidb/pull/27741)

File: releases/release-4.0.15.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+        - 修复了备份和恢复中平均速度计算不准确的问题 [#1405](https://github.com/pingcap/br/issues/1405)
+
+    + TiCDC
+
+        - 修复集成测试中遇到的由于 DDL Job 重复导致的 `ErrSchemaStorageTableMiss` 错误 [#2422](https://github.com/pingcap/ticdc/issues/2422)
+        - 修复遇到 `ErrGCTTLExceeded` 错误时 changefeed 无法被删除的问题 [#2391](https://github.com/pingcap/ticdc/issues/2391)
+        - 修复 `capture list` 命令输出中出现已过期 capture 的问题 [#2388](https://github.com/pingcap/ticdc/issues/2388)
+        - 修复 TiCDC processor 出现死锁的问题 [#2017](https://github.com/pingcap/ticdc/pull/2017)
+        - 修复重新调度一张表时多个处理器将数据写入同一张表引发的数据不一致的问题 [#2230](https://github.com/pingcap/ticdc/issues/2230) [#2230](https://github.com/pingcap/ticdc/issues/2230)
```
Comment (TomShawn) [inline releases/release-4.0.15.md:126]:
```suggestion
        - 修复重新调度一张表时多个处理器将数据写入同一张表引发的数据不一致的问题 [#2230](https://github.com/pingcap/ticdc/issues/2230)
```

**Comment by disksing** [comment]
LGTM

File: releases/release-4.0.15.md [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+        - 修复集成测试中遇到由于 DDL Job 重复导致的 `ErrSchemaStorageTableMiss` 错误 [#2422](https://github.com/pingcap/ticdc/issues/2422)
+        - 修复遇到 `ErrGCTTLExceeded` 错误时 changefeed 无法被删除的问题 [#2391](https://github.com/pingcap/ticdc/issues/2391)
+        - 修复 `capture list` 命令输出中出现已过期 capture 的问题 [#2388](https://github.com/pingcap/ticdc/issues/2388)
+        - 修复 TiCDC processor 出现死锁的问题 [#2017](https://github.com/pingcap/ticdc/pull/2017)
+        - 修复重新调度一张表时多个处理器将数据写入同一张表引发的数据不一致的问题 [#2230](https://github.com/pingcap/ticdc/issues/2230)
+        - 修复元数据管理出现 `EtcdWorker` 快照隔离被破坏的问题 [#2557](https://github.com/pingcap/ticdc/pull/2557)
+        - 修复因为 DDL sink 错误导致 changefeed 不能被停止的问题 [#2552](https://github.com/pingcap/ticdc/issues/2552)
+        - 修复一个 TiCDC Open Protocol 的问题：当一个事务中没有任何数据写入时候，TiCDC 产生一个空消息 [#2612](https://github.com/pingcap/ticdc/issues/2612)
+        - 修复 TiCDC 在处理无符号 `TINYINT` 类型时崩溃的问题 [#2648](https://github.com/pingcap/ticdc/issues/2648)
+        - 修复内存压力大时 gRPC 连接频繁断开的错误 [#2202](https://github.com/pingcap/ticdc/issues/2202)
```
Comment (TomShawn) [inline releases/release-4.0.15.md:131]:
```suggestion
        - 减小 gRPC 窗口来避免 Region 数量过多时触发内存溢出 [#2202](https://github.com/pingcap/ticdc/issues/2202)
```

**Comment by IANTHEREAL** [comment]
LGTM

**Comment by guo-shaoge** [comment]
LGTM

**Comment by ethercflow** [comment]
LGTM

**Review by Liuxiaozhen12 — APPROVED** [review]

**Pull request merged**